File: src/main/java/com/hellosign/sdk/resource/support/types/FieldType.java
package com.hellosign.sdk.resource.support.types;

/**
 * Enumeration of allowable field types. These are used in the ResponseData and FormField objects.
 */
public enum FieldType {
    CHECKBOX("checkbox"),
    CHECKBOX_MERGE("checkbox-merge"),
    DATE_SIGNED("date_signed"),
    DROPDOWN("dropdown"),
    INITIALS("initials"),
    RADIO("radio"),
    SIGNATURE("signature"),
    TEXT("text"),
    TEXT_MERGE("text-merge");

    public final String value;

    FieldType(String value) {
        this.value = value;
    }

    public static FieldType getEnum(String string) {
        for (FieldType type : values()) {
            if (type.value.equals(string)) {
                return type;
            }
        }
        throw new IllegalArgumentException();
    }

    @Override
    public String toString() {
        return this.value;
    }
}

File: src/main/java/com/hellosign/sdk/resource/support/types/FieldType.kt
package com.hellosign.sdk.resource.support.types

import java.lang.IllegalArgumentException

/**
 * Enumeration of allowable field types. These are used in the ResponseData and FormField objects.
 */
enum class FieldType(val value: String) {
    CHECKBOX("checkbox"),
    CHECKBOX_MERGE("checkbox-merge"),
    DATE_SIGNED("date_signed"),
    DROPDOWN("dropdown"),
    INITIALS("initials"),
    RADIO("radio"),
    SIGNATURE("signature"),
    TEXT("text"),
    TEXT_MERGE("text-merge");

    override fun toString(): String {
        return value
    }

    companion object {
        @JvmStatic
        fun getEnum(value: String): FieldType? {
            return enumValues<FieldType>().find { it.value === value }
        }
    }
}
